eng/tools/TypeSpecValidation/src/rules/emit-autorest.ts

```diff
@@ -0,0 +1,45 @@
+import { readFile } from "fs/promises";
+import { join } from "path";
+import { parse as yamlParse } from "yaml";
+import { Rule } from "../rule.js";
+import { RuleResult } from "../rule-result.js";
+import { checkFileExists } from "../utils.js";
+
+export class EmitAutorestRule implements Rule {
+  readonly name = "EmitAutorest";
+
+  readonly description = 'Must emit "@azure-tools/typespec-autorest" by default';
+
+  async execute(folder: string): Promise<RuleResult> {
+    let success = true;
+    let stdOutput = "";
+    let errorOutput = "";
+
+    const mainTspExists = await checkFileExists(join(folder, "main.tsp"));
+    stdOutput += `mainTspExists: ${mainTspExists}`;
+
+    if (mainTspExists) {
+      const configFile = join(folder, "tspconfig.yaml");
+      const configText = await readFile(configFile, "utf8");
+      const config = yamlParse(configText);
+
+      const emit = config.emit;
+      stdOutput += `emit: ${JSON.stringify(emit)}\n`;
+
+      if (!emit?.includes("@azure-tools/typespec-autorest")) {
```

Comment:
what if there's no emit at all (emit === null | undefined)

Reply:
If `emit` is `null` or `undefined`, the `?.` operator will short-circuit and return `undefined`, which evaluates to `false`, so the rule will return an error.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining